src/subscription/subscribe.js

```diff
@@ -106,16 +110,33 @@ function subscribeImpl(
   fieldResolver,
   subscribeFieldResolver
 ) {
-  const subscription = createSourceEventStream(
+  // If arguments are missing or incorrect, throw an early error.
+  assertValidExecutionArguments(
```

Comment:
This can throw. If we want to be consistent, should this be changed to also yield a rejected async iterator?